homeassistant/components/motion_blinds/__init__.py

```diff
@@ -106,40 +107,46 @@ async def _async_update_data(self):
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the motion_blinds components from a config entry."""
     hass.data.setdefault(DOMAIN, {})
+    setup_lock = hass.data[DOMAIN].setdefault(KEY_SETUP_LOCK, asyncio.Lock())
     host = entry.data[CONF_HOST]
     key = entry.data[CONF_API_KEY]
     multicast_interface = entry.data.get(CONF_INTERFACE, DEFAULT_INTERFACE)
     wait_for_push = entry.options.get(CONF_WAIT_FOR_PUSH, DEFAULT_WAIT_FOR_PUSH)
 
     entry.async_on_unload(entry.add_update_listener(update_listener))
 
-    # check multicast interface
-    check_multicast_class = ConnectMotionGateway(hass, interface=multicast_interface)
-    working_interface = await check_multicast_class.async_check_interface(host, key)
-    if working_interface != multicast_interface:
-        data = {**entry.data, CONF_INTERFACE: working_interface}
-        hass.config_entries.async_update_entry(entry, data=data)
-        _LOGGER.debug(
-            "Motion Blinds interface updated from %s to %s, "
-            "this should only occur after a network change",
-            multicast_interface,
-            working_interface,
-        )
-
     # Create multicast Listener
-    if KEY_MULTICAST_LISTENER not in hass.data[DOMAIN]:
-        multicast = AsyncMotionMulticast(interface=working_interface)
-        hass.data[DOMAIN][KEY_MULTICAST_LISTENER] = multicast
-        # start listening for local pushes (only once)
-        await multicast.Start_listen()
-
-        # register stop callback to shutdown listening for local pushes
-        def stop_motion_multicast(event):
-            """Stop multicast thread."""
-            _LOGGER.debug("Shutting down Motion Listener")
-            multicast.Stop_listen()
-
-        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_motion_multicast)
+    async with setup_lock:
+        if KEY_MULTICAST_LISTENER not in hass.data[DOMAIN]:
```

Comment:
Invert the check and return if true. Then we can outdent below.

Reply:
Do you mean with a `break` statement?
I am not sure if I can use `break` for a `async with setup_lock`?
The code below the setup_lock still needs to be run regardless of the outcome of the check.

Reply:
Sorry, I thought we were inside a separate function that we could return early from.

Reply:
No worries